src/stellar_format.c

```diff
@@ -744,19 +817,38 @@ static void format_path_payment_strict_receive(tx_context_t *txCtx) {
     push_to_formatter_stack(&format_path_payment_strict_receive_send);
 }
 
+static void format_path_payment_strict_send_path_via(tx_context_t *txCtx) {
+    strcpy(detailCaption, "Via");
+    uint8_t i;
+    for (i = 0; i < txCtx->txDetails.opDetails.pathPaymentStrictSendOp.pathLen; i++) {
+        char asset_name[12 + 1];
+        Asset *asset = &txCtx->txDetails.opDetails.pathPaymentStrictSendOp.path[i];
+        if (strlen(detailValue) != 0) {
+            strlcat(detailValue, ", ", DETAIL_VALUE_MAX_SIZE);
+        }
+        print_asset_name(asset, txCtx->network, asset_name, sizeof(asset_name));
+        strlcat(detailValue, asset_name, DETAIL_VALUE_MAX_SIZE);
+    }
+    format_operation_source_prepare(txCtx);
+}
+
 static void format_path_payment_strict_send_receive(tx_context_t *txCtx) {
     strcpy(detailCaption, "Receive Min");
-    print_amount(txCtx->txDetails.opDetails.pathPaymentStrictReceiveOp.destAmount,
```

Comment:
Since pathPaymentStrictReceiveOp and pathPaymentStrictSendOp have the same memory layout, they can actually display correct data. But we are here to fix it.